geode-core/src/main/java/org/apache/geode/pdx/internal/json/PdxToJSON.java

```diff
@@ -135,17 +135,16 @@ private void writeValue(JsonGenerator jg, Object value, String pf)
       jg.writeString(value.toString());
     } else if (value.getClass().equals(PdxInstanceEnumInfo.class)) {
       jg.writeString(value.toString());
+    } else if (value instanceof PdxInstance) {
+      getJSONString(jg, (PdxInstance) value);
+    } else if (value instanceof Collection) {
+      getJSONStringFromCollection(jg, (Collection<?>) value, pf);
+    } else if (value instanceof Map) {
+      getJSONStringFromMap(jg, (Map) value, pf);
     } else {
-      if (value instanceof PdxInstance) {
-        getJSONString(jg, (PdxInstance) value);
-      } else if (value instanceof Collection) {
-        getJSONStringFromCollection(jg, (Collection<?>) value, pf);
-      } else if (value instanceof Map) {
-        getJSONStringFromMap(jg, (Map) value, pf);
-      } else {
-        throw new IllegalStateException(
-            "PdxInstance returns unknwon pdxfield " + pf + " for type " + value);
-      }
+      throw new IllegalStateException(
+          "PdxInstance returns unsupported type " + value.getClass()
```

Comment:
I think this message should make clear what we mean by "unsupported". Pdx supports it but we don't support converting it to JSON. So something like: "The pdx field 'FIELD_NAME' has a value whose type 'TYPE' can not be converted to JSON. The value is 'VALUE'."
You need to add a unit test for this message

Reply:
Agreed, though I went with a single sentence:
`          "The pdx field " + pf + " has a value " + value + " whose type " + value.getClass()
              + " can not be converted to JSON.");
`

Reply:
nice